### PR TITLE
Add Pronic Number Implementation

### PR DESCRIPTION
**Pull request by itsAkshayDubey**
### **Describe your change:**

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

#### References

Pronic Number Implemenation #2856 

### **Checklist:**

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new Java files are placed inside an existing directory.
- [x] All filenames are in all uppercase characters with no spaces or dashes.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by siriak — CHANGES_REQUESTED** [review]
Looks good, could you add a couple of tests to make sure everything works as in https://github.com/TheAlgorithms/Java/blob/66d8d51de67fd3e7c7249930f4a912ad9560d823/src/test/java/com/thealgorithms/searches/QuickSelectTest.java?

**Comment by itsAkshayDubey** [comment]
Hi @siriak , added unit tests for Pronic Number implementation. Kindly have a look.

File: src/main/java/com/thealgorithms/maths/PronicNumber.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	
+	//Driver code
+	public static void main(String[] args) {
+		
+		System.out.println(isPronic(30));
+		
+		//Ouput: true
+	}
+	
+}
```
Comment (siriak) [inline src/main/java/com/thealgorithms/maths/PronicNumber.java:47]:
```suggestion
	}
}
```

**Review by siriak — APPROVED** [review]
LGTM, thanks!

**Pull request merged**